src/main/java/bio/terra/service/search/SearchService.java

```diff
@@ -46,12 +44,14 @@
     private final BigQueryPdao bigQueryPdao;
     private final RestHighLevelClient client;
 
+    @Value("${elasticsearch.numShards}")
+    private static int NUM_SHARDS;
+
     @Autowired
-    public SearchService(BigQueryPdao bigQueryPdao) {
-        // needs to be moved into own class with config
-        final RestClientBuilder builder = RestClient.builder(new HttpHost("localhost", 9200));
+    public SearchService(BigQueryPdao bigQueryPdao, RestHighLevelClient client) {
```

Comment:
This is interesting; so because there's a constructor `RestHighLevelClient(RestClientBuilder restClientBuilder)` and `RestClientBuilder` is injectable, you can inject `RestHighLevelClient` here?

Is there a reason why you didn't make `RestHighLevelClient` itself injectable? That seems like it would work too, and I think it would make the behavior of this code clearer (to me, at least).

Reply:
If I make my own @Bean for RestHighLevelClient, it complains that there's an existing @Bean for RestHighLevelClient declared in ElasticSearchRestClientConfigurations.java, which is a spring boot autoconfigure class.

So the RestHighLevelClient is a bean with a RestClientBuilder as a param, if I make a RestClientBuilder bean, it will inject into RestHighLevelClient automatically.